src/main/java/elrond/MessageSigning.java

```diff
@@ -0,0 +1,72 @@
+package elrond;
+
+import org.bouncycastle.crypto.digests.KeccakDigest;
+import org.bouncycastle.crypto.params.Ed25519PrivateKeyParameters;
+import org.bouncycastle.crypto.params.Ed25519PublicKeyParameters;
+import org.bouncycastle.crypto.signers.Ed25519Signer;
+import org.bouncycastle.util.encoders.Hex;
+
+public class MessageSigning {
+    static final String signerMessagePrefix = "\\x17Elrond Signed Message:\n";
+    private KeccakDigest hasher;
+    private byte[] privateKey;
+
+    public MessageSigning(String privateKeyHex) {
+        this(Hex.decode(privateKeyHex));
+    }
+
+    public MessageSigning(byte[] privateKey) {
+        this.privateKey = privateKey;
+        this.hasher = createKeccakHasher();
+    }
+
+    public String SignMessage(byte[] msg) {
+        Ed25519Signer signer = createEd25519Signer();
+        byte[] message = computeHashOnMessage(msg);
+
+        signer.update(message, 0, message.length);
```

Comment:
couldn't you use `Wallet` class? so you don't have to duplicate the signing logic

Reply:
i was thinking maybe the other way, to try to use message signing in wallet, but without custom hashing